File: source/code-snippets/usage-examples/updateOne.js
// ignored first line
const { MongoClient } = require("mongodb");

// Replace the uri string with your MongoDB deployment's connection string.
const uri =
  "mongodb+srv://<user>:<password>@<cluster-url>?w=majority";

const client = new MongoClient(uri);

async function run() {
  try {
    await client.connect();

    const database = client.db("sample_mflix");
    const collection = database.collection("movies");

    // create a filter for a movie to update
    const filter = { title: "Blacksmith Scene" };
    const options = {};
    options.upsert = true; // create a document if no documents match the filter
    // create a document that sets the plot of the movie
    const updateDoc = {
      $set: {
        plot:
          "Blacksmith Scene is a silent film directed by William K.L. Dickson",
      },
    };
    const result = await collection.updateOne(filter, updateDoc, options);
    console.log(
      `${result.matchedCount} document matched the filter and ${result.modifiedCount} document was updated`,
    );
  } finally {
    await client.close();
  }
}
run().catch(console.dir);

File: source/code-snippets/updateOne.go
package main

import (
	"context"
	"fmt"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

// Replace the uri string with your MongoDB deployment's connection string.
const uri = "mongodb+srv://<username>:<password>@<cluster-address>/test?w=majority"

func main() {
	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))

	if err != nil {
		panic(err)
	}
	defer func() {
		if err = client.Disconnect(context.TODO()); err != nil {
			panic(err)
		}
	}()

	// begin updateone
	myCollection := client.Database("sample_restaurants").Collection("restaurants")
	id, _ := primitive.ObjectIDFromHex("5eb3d668b31de5d588f42a7a")
	filter := bson.D{{"_id", id}}
	update := bson.D{{"$set", bson.D{{"avg_rating", 4.4}}}}

	result, err := myCollection.UpdateOne(context.TODO(), filter, update)
	// end updateone

	if err != nil {
		panic(err)
	}

	fmt.Printf("Documents updated: %v\n", result.ModifiedCount)
}
